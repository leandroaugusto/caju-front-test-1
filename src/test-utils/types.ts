export interface ITestUtils {
  children?: React.ReactNode
}
