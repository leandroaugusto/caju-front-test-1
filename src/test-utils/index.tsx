import { ReactElement } from "react";
import { ThemeProvider } from "styled-components";
import { render, RenderOptions, RenderResult } from "@testing-library/react";

import { ITestUtils } from "./types";

import theme from "~/theme";

const AllTheProviders: React.FC = ({ children }: ITestUtils) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
