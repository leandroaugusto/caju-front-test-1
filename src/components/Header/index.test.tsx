import { customRender, screen } from "~/test-utils";

import { Header } from ".";

describe("Header", () => {
  it("Should render header component", () => {
    const { container } = customRender(
      <Header data-testid="header">
        <h1 data-testid="heading-title">Caju Front Teste</h1>
      </Header>
    );
    const headingTitle = screen.getByTestId("heading-title");

    expect(headingTitle).toHaveTextContent("Caju Front Teste");
    expect(container).toMatchSnapshot();
  });
});
