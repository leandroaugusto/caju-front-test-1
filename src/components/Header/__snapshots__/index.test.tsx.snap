// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Should render header component 1`] = `
<div>
  <header
    class="sc-braxZu Ghofx"
    data-testid="header"
  >
    <h1
      data-testid="heading-title"
    >
      Caju Front Teste
    </h1>
  </header>
</div>
`;
